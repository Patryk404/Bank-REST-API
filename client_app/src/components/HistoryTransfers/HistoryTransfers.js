import React from 'react';
import classes from './HistoryTransfers.module.css';
import HistoryTransfer from './HistoryTransfer/HistoryTransfer';
import Aux from '../../hoc/Auxiliary/Auxiliary';
const historyTransfers = (props)=>{
    let transfers;
    if (props.history)
    {
        transfers = props.history.map(
            (element,index)=>{// setting list with appropriate key
                return (
                <HistoryTransfer key={index}>{element}</HistoryTransfer>
                );
            }
        )
    }
    else { 
        transfers = null;
    }
    return(
        <Aux>
            <div className={classes.Transfers}>
                <h1 style={{fontSize: '20px',fontWeight: 'normal'}}>History of Transfers</h1>
                    <ul>
                        {transfers}
                    </ul>
            </div>
        </Aux>
    );
}; 

export default historyTransfers;