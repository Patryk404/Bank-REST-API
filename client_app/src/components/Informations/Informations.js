import React from 'react';
import classes from './Informations.module.css';
const informations = (props)=>{
    return(
        <div className={classes.Informations}>
            <h1>Informations</h1>
            <ul>
            <li><h2 style={{fontSize: '36px'}}>{props.user.name} {props.user.surname}</h2></li>
            <li><h3>Cash: {props.user.cash}</h3></li>
            <li><h3>Bill Number: {props.user.bill}</h3></li>
            </ul>
        </div>
    );
}

export default informations;