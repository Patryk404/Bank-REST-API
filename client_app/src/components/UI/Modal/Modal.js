import React from 'react';
import classes from './Modal.module.css';
import Backdrop from '../Backdrop/Backdrop';
import Aux from '../../../hoc/Auxiliary/Auxiliary';
const modal = (props)=> {
    return (
        <Aux>
            <Backdrop show={props.show} click={props.clickonbackdrop}/>
            <div className={classes.Modal}
                style={{
                    transform: props.show ? 'translateY(0)' : 'translateY(-100vh)',//animtion if our props show is true or not 
                    opacity: props.show ? '1' : '0' // visibility 
               }}>
               {props.children}
            </div>
        </Aux>
    );
}

export default modal;