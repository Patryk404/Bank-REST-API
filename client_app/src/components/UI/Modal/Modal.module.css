.Modal {
    position: fixed;
    z-index: 500;
    background-color:  #c2f0f0;
    width: 70%;
    border: 3px solid black;
    box-shadow: 1px 1px 1px black;
    padding: 16px;
    left: 15%;
    top: 30%;
    box-sizing: border-box;
}
.Modal-enter {
    opacity: 0;
    transform: translateY(-100vh);
    position: fixed;
    z-index: 500;
    background-color:  #c2f0f0;
    width: 70%;
    border: 3px solid black;
    box-shadow: 1px 1px 1px black;
    padding: 16px;
    left: 15%;
    top: 30%;
    box-sizing: border-box;
    transition: all 0.3s ease-out;
}

.Modal-enter-active {
    opacity: 1;
    transform: translateY(0);
    position: fixed;
    z-index: 500;
    background-color:  #c2f0f0;
    width: 70%;
    border: 3px solid black;
    box-shadow: 1px 1px 1px black;
    padding: 16px;
    left: 15%;
    top: 30%;
    box-sizing: border-box;
    transition: all 0.3s ease-out;
}

.Modal-exit-active {
    opacity: 1;
    transform: translateY(-100vh);
    position: fixed;
    z-index: 500;
    background-color:  #c2f0f0;
    width: 70%;
    border: 3px solid black;
    box-shadow: 1px 1px 1px black;
    padding: 16px;
    left: 15%;
    top: 30%;
    box-sizing: border-box;
    transition: all 0.3s ease-out;
}

.Modal-exit {
    opacity: 0;
    transform: translateY(-100vh);
    position: fixed;
    z-index: 500;
    background-color:  #c2f0f0;
    width: 70%;
    border: 3px solid black;
    box-shadow: 1px 1px 1px black;
    padding: 16px;
    left: 15%;
    top: 30%;
    box-sizing: border-box;
    transition: all 0.3s ease-out;
}

@media (min-width: 600px) {
    .Modal {
        width: 500px;
        left: calc(50% - 250px);
    }
}