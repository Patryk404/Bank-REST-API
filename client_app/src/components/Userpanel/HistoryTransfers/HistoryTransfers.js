import React from 'react';
import classes from './HistoryTransfers.module.css';
import HistoryTransfer from './HistoryTransfer/HistoryTransfer';
import Aux from '../../../hoc/Auxiliary/Auxiliary';
const historyTransfers = (props)=>{
    const transfers = props.history.map(
        (element,index)=>{// setting list with appropriate key
            return (
            <HistoryTransfer key={index}>{element}</HistoryTransfer>
            );
        }
    )
    return(
        <Aux>
            <div className={classes.Transfers}>
                <h1 style={{fontSize: '20px',fontWeight: 'normal'}}>History of Transfers</h1>
                    <ul>
                        {transfers}
                    </ul>
            </div>
        </Aux>
    );
}; 

export default historyTransfers;