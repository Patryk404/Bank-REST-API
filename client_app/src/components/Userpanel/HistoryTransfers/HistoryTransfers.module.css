.Transfers {
    height: 60%; /* 100% Full-height */
    width: 20%; /* 0 width - change this with JavaScript */
    position: fixed; /* Stay in place */
    top: 150px; /* Stay at the top */
    left: 40px;
    background-color: #58eddc;
    border-radius: 10px;
    overflow-y: auto; /* Disable horizontal scroll */
    padding-top: 15px; /* Place content 60px from the top */
    transition: 0.5s; /* 0.5 second transition effect to slide in the sidenav */
    list-style-type: none;
    text-align: center;
}
.Transfers ul{
    display: inline;
}