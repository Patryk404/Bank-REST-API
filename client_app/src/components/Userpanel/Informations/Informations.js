import React from 'react';
import classes from './Informations.module.css';
const informations = (props)=>{
    return(
        <div className={classes.Informations}>
            <h1>Informations</h1>
            <ul>
            <li><h2>{props.name_and_surname}</h2></li>
            <li><h3>Cash: {props.cash}</h3></li>
            </ul>
        </div>
    );
}

export default informations;