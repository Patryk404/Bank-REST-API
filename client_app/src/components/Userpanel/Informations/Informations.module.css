.Informations {
    position: fixed;
    bottom:  300px;
    left: 500px;
    display: inline-block;
    text-align: center;
    background-color: #ABFFFF;
    border: 1px solid black;
    width: 50%;
}

.Informations ul {
    list-style-type: none;
    display: inline;
}