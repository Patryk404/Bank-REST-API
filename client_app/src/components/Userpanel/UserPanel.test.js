import UserPanel from './UserPanel';
import HistoryTransfers from '../Userpanel/HistoryTransfers/HistoryTransfers';
import React from 'react';
import Aux from '../../hoc/Auxiliary/Auxiliary';
import Layout from '../../hoc/Layout/Layout';
import Informations from '../../components/Userpanel/Informations/Informations';
import MakeTransfer from './Maketransfer/Maketransfer';

import {configure,shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Modal from '../UI/Modal/Modal';

configure({adapter: new Adapter});

describe('<UserPanel />',()=>{
    let wrapper;
    beforeEach(()=>{
        wrapper = shallow(<UserPanel/>);
        localStorage.setItem('token','6845976957468945');
    })

    it('matches snapshot',()=>{
        wrapper = renderer.create(<UserPanel/>).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    it('should render <HistoryTransfers> component',()=>{
        wrapper.setState({loggedUser:{
            transfers:[]
        }});
        expect(wrapper.find(HistoryTransfers)).toHaveLength(1);
    });
    it('should render modal with MakeTransfer if we click into button ',()=>{
        expect(wrapper.state().showMakeTransfer).toEqual(false);
        const ButtonComponent = wrapper.find('button').first();
        ButtonComponent.props().click();
        //console.log('find button component', ButtonComponent.debug());
        //wrapper.instance().makeTransferButtonHandler();
        expect(wrapper.find(Modal)).toHaveLength(1);
    })
    it('should not render modal if we not click on button make_transfer',()=>{
        wrapper.setState({showMakeTransfer: false});
        expect(wrapper.find(Modal)).toHaveLength(0); 
    })
    it('should render <Aux> component',()=>{
        expect(wrapper.find(Aux)).toHaveLength(1);
    })
    it('should render <Layout> component',()=>{
        expect(wrapper.find(Layout)).toHaveLength(1);
    })
    it('should render <Informations> component',()=>{
        expect(wrapper.find(Informations)).toHaveLength(1);
    })
    it('should not render <MakeTransfer> component',()=>{
        expect(wrapper.find(MakeTransfer)).toHaveLength(0);
    })
    it('should render <MakeTransfer> component if stateshowMakeTransfer is true',()=>
    {
        wrapper.setState({showMakeTransfer: true});
        expect(wrapper.find(MakeTransfer)).toHaveLength(1);
    })
    it('should render h1 if we are not authenticated',()=>{
        localStorage.removeItem('token');
        expect(wrapper.find('h1')).toHaveLength(1);
    })
    
})