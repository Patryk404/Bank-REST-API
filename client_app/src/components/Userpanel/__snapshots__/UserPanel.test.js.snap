// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserPanel /> matches snapshot 1`] = `
Array [
  <div
    className="Logo_logged"
  >
    <img
      alt="Logo"
      src="logo.png"
    />
  </div>,
  <h1
    style={
      Object {
        "fontWeight": "normal",
      }
    }
  >
    Panel
  </h1>,
  <div
    style={
      Object {
        "position": "absolute",
        "top": "200px",
        "width": "100%",
      }
    }
  >
    <button
      className="Button"
      onClick={[Function]}
    >
      Make Transfer
    </button>
  </div>,
  <div
    style={
      Object {
        "left": "800px",
        "position": "fixed",
        "textAlign": "none",
        "top": "0px",
        "width": "75%",
      }
    }
  >
    <button
      className="Button Red"
      onClick={[Function]}
    >
      Log Out
    </button>
  </div>,
  <div
    className="Transfers"
  >
    <h1
      style={
        Object {
          "fontSize": "20px",
          "fontWeight": "normal",
        }
      }
    >
      History of Transfers
    </h1>
    <ul />
  </div>,
  <div
    className="Informations"
  >
    <h1>
      Informations
    </h1>
    <ul>
      <li>
        <h2
          style={
            Object {
              "fontSize": "36px",
            }
          }
        >
          
           
          
        </h2>
      </li>
      <li>
        <h3>
          Cash: 
        </h3>
      </li>
      <li>
        <h3>
          Bill Number: 
          
        </h3>
      </li>
    </ul>
  </div>,
]
`;
